pkg/webhook/resources/setting/validator.go

```diff
@@ -1142,6 +1167,71 @@ func (v *settingValidator) checkStorageNetworkValueVaild() error {
 	return nil
 }
 
+func (v *settingValidator) checkClusterNetworkReady(config *storagenetworkctl.Config) error {
+	cn, err := v.cnCache.Get(config.ClusterNetwork)
+	if err != nil {
+		return err
+	}
+
+	if networkv1.Ready.IsFalse(cn.Status) {
+		return fmt.Errorf("cluster network %s is not Ready", config.ClusterNetwork)
+	}
+
+	return nil
+}
+
+func getMatchNodes(vc *networkv1.VlanConfig) ([]string, error) {
+	if vc.Annotations == nil || vc.Annotations[utils.KeyMatchedNodes] == "" {
+		return nil, nil
+	}
+
+	var matchedNodes []string
+	if err := json.Unmarshal([]byte(vc.Annotations[utils.KeyMatchedNodes]), &matchedNodes); err != nil {
+		return nil, err
+	}
+
+	return matchedNodes, nil
+}
+
+func (v *settingValidator) checkVCSpansAllNodes(config *storagenetworkctl.Config) error {
+	var matchedNodes mapset.Set[string]
+	var vlanConfigNodes []string
+
+	vcs, err := v.vcCache.List(labels.Set{
+		utils.KeyClusterNetworkLabel: config.ClusterNetwork,
+	}.AsSelector())
+	if err != nil {
+		return err
+	}
+
+	if len(vcs) > 0 {
+		for _, vc := range vcs {
+			vnodes, err := getMatchNodes(vc)
+			if err != nil {
+				return err
+			}
+			vlanConfigNodes = append(vlanConfigNodes, vnodes...)
+		}
+		matchedNodes = mapset.NewSet[string](vlanConfigNodes...)
+	} else {
+		return fmt.Errorf("vlan config is not present for cluster network %s", config.ClusterNetwork)
+	}
+
+	nodes, err := v.nodeCache.List(labels.Everything())
+	if err != nil {
+		return werror.NewInternalError(err.Error())
+	}
+
+	//check if vlanconfig contains all the nodes in the cluster
```

Comment:
The `witness-node` needs to be skipped https://docs.harvesterhci.io/v1.4/advanced/witness. It does not deploy LH related pods or VMs. 

Another point:

After storage-network is enabled, a new none-witness node is added, and this node is not covered by VC, how can we hint this?

Reply:
Sure, will add the witness-node check to the changes.

I think pkg/webhook/resource/node/validator.go can handle checks during Update where a new node addition is handled.
The storage network nad cache could be validated to see if it exists but the challenge is webhook returns always an "error" or nil.Is there something where a message could be notified to the user ? Need to check with someone from UI if there is any other way any info from the cluster could be notified to the user/UI.

Reply:
Added the code changes to skip "witness-node" during the validation

Reply:
One solution is to add metrics and alerts like https://kubevirt.io/monitoring/metrics.html#kubevirt_virt_controller_ready, https://kubevirt.io/monitoring/runbooks/NoReadyVirtOperator.html to indicate that certain nodes in the storagenetwork are not ready.

We may design and implement this feature in v1.5.0. https://github.com/harvester/harvester/issues/2850